pages/instructorQuestionManualGrading/instructorQuestionManualGradingNextInstanceQuestion.sql

```diff
@@ -1,16 +1,27 @@
 -- BLOCK get_next_ungraded_instance_question
-SELECT * FROM (SELECT DISTINCT ON (iq.id)
-    iq.*,
-    s.graded_at,
-    ai.id AS assessment_instance_id
-FROM
-    instance_questions AS iq
-    JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
-    JOIN variants AS v ON (v.instance_question_id = iq.id)
-    JOIN submissions AS s ON (s.variant_id = v.id)
-WHERE
-    ai.assessment_id = $assessment_id
-    AND iq.assessment_question_id = $assessment_question_id
-ORDER BY iq.id DESC, s.date DESC, s.id DESC) AS submission_info
-WHERE submission_info.graded_at IS NULL
+WITH to_grade AS (
+    SELECT *, (lead(id) OVER (ORDER BY user_or_group_name DESC) IS NOT DISTINCT FROM $prior_instance_question_id) AS is_next
+    FROM (
+        SELECT DISTINCT ON (iq.id)
+            iq.*,
+            s.graded_at,
+            ai.id AS assessment_instance_id,
+            COALESCE(g.name, u.name) AS user_or_group_name
+        FROM
+            instance_questions AS iq
+            JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
+            JOIN variants AS v ON (v.instance_question_id = iq.id)
+            JOIN submissions AS s ON (s.variant_id = v.id)
+            LEFT JOIN users u ON (u.user_id = ai.user_id)
+            LEFT JOIN groups g ON (g.id = ai.group_id)
+        WHERE
+            ai.assessment_id = $assessment_id
+            AND iq.assessment_question_id = $assessment_question_id
+        ORDER BY iq.id DESC, s.date DESC, s.id DESC) AS submission_info
+    WHERE submission_info.graded_at IS NULL OR submission_info.id = $prior_instance_question_id
+)
+SELECT * FROM to_grade
+WHERE graded_at IS NULL
```

Comment:
After the change in #5103, will manually grading a question actually set `graded_at` on the submission? I worry that it won't, so here we'll just keep selecting these questions.

Reply:
It is, right here: 
https://github.com/PrairieLearn/PrairieLearn/blob/0473cb9a848c5b93f3803f371da83fc12e7eac81/sprocs/instance_questions_update_score.sql#L126-L151

I just noticed by looking there, however, that they are also being marked as `grading_method='External'`, which isn't really correct.

Reply:
Ok, great. I see we _are_ correctly marking the `grading_job` as `grading_method='Manual'`. Want to fix the `External`?

Reply:
Wait, this same function call has been for instructors manually editing grades for ages. Has this been wrong for that case all along too?

Reply:
I think so? I don't think we use `grading_method` for anything important, so it hasn't been causing a problem?

Reply:
I guess. Let's just open a new issue for it.

Reply:
https://github.com/PrairieLearn/PrairieLearn/issues/5110